src/Storages/ObjectStorage/DataLakes/DataLakeConfiguration.h

```diff
@@ -329,10 +349,69 @@ class DataLakeConfiguration : public BaseStorageConfiguration, public std::enabl
         current_metadata->addDeleteTransformers(object_info, builder, format_settings, local_context);
     }
 
+    void fromDisk(const String & disk_name, ASTs & args, ContextPtr context, bool with_structure) override
+    {
+        if (!Context::getGlobalContextInstance()->getAllowedDisksForTableEngines().contains(disk_name))
+            throw Exception(ErrorCodes::BAD_ARGUMENTS, "Disk {} is not allowed for usage in storage engines", disk_name);
+
+        BaseStorageConfiguration::fromDisk(disk_name, args, context, with_structure);
+        auto disk = context->getDisk(disk_name);
+        ready_object_storage = disk->getObjectStorage();
+
+        if (args.size() > 2 + with_structure)
```

Comment:
What do you think about moving this code of parsing `format`, `structure`, etc, into `Utils.cpp` and using not from here (`DataLakeConfiguration::fromDisk`), but from `BaseStorageConfiguration::fromDisk(disk_name, args, context, with_structure);`? The motivation for this is that if you look at concrete method `S3::Configuration::fromDisk` - this method is incorrect - because it does not parse `format`/etc arguments, while all other such methods - `fromAST`, `fromNamedCollection` - correctly process all the arguments. E.g. this will be more correct, while putting this part of code into `Utils.cpp` will allow to avoid copy-paste.